### ISSUE TITLE
Explorer - Commands: Move 'isExplorerActive' filter to command registration

### ISSUE THREAD

**Issue by bryphe** [onivim/oni#2194]
Porting this out from the conversation in #2115 from @CrossR :

> That said, I think we need to improve the filtering for user binds. As you can see in the default binds, the binds only work when the explorer is active, so we should probably offer an easy way to replicate that functionality in the user config, since currently I think its a bit of a pain.

It's hard to reason about how to work with the `isExplorerActive` filter, if you decide you want to remap the keybindings:
https://github.com/onivim/oni/blob/75e2b74770bdbcbb884af348b05de19e2f6d45f1/browser/src/Input/KeyBindings.ts#L52

If someone wants to customize the explorer keybindings, they'll have to copy & paste this over to their `config.tsx`, which is painful. If they don't do this, they could run into issues like:  #2162 Not very ergonomic.

The `commands` have the option of specifying a filter, too. When we evaluate an input character, we first  look at the filter provided by the user `bind`, and then we look at the `enabled` function on the `command` (if its a string command that's passed). That's really the one advantage of having a string `command` registered vs just having a function - when registering this string `command`, an `enabled` function can be optionally provided to specify when the command when should be active. This is a perfect use case of leveraging that `enabled` filter so that the user doesn't need tow orry about it.

So to make this more ergonomic, so that the user doesn't need to know to add `isExplorerActive` to all the explorer bindings, we should push that `isExplorerActive` filter over to where we register the commands in `ExplorerSplit.tsx`: 
https://github.com/onivim/oni/blob/75e2b74770bdbcbb884af348b05de19e2f6d45f1/browser/src/Services/Explorer/ExplorerSplit.tsx#L124

If we add that `isExplorerActive` predicate to the filter for the commands, then the user will just be able to do:
```
input.bind("r", "explorer.rename")
```

If the `enabled` filter returns `false` for `explorer.rename`, then that will be skipped / not evaluated.

__TL;DR:__ We should move the `isExplorerActive` filter from the default key bindings to the command registration, so that users don't need to provide a filter when binding / rebinding the explorer split commands.
